cmd/cluster-agent-cloudfoundry/app/app.go

```diff
@@ -0,0 +1,261 @@
+		logFile,
+		syslogURI,
+		config.Datadog.GetBool("syslog_rfc"),
+		config.Datadog.GetBool("log_to_console"),
+		config.Datadog.GetBool("log_format_json"),
+	)
+	if err != nil {
+		log.Criticalf("Unable to setup logger: %s", err)
+		return nil
+	}
+
+	if !config.Datadog.IsSet("api_key") {
+		log.Critical("no API key configured, exiting")
+		return nil
+	}
+
+	// Setup healthcheck port
+	var healthPort = config.Datadog.GetInt("health_port")
+	if healthPort > 0 {
+		err := healthprobe.Serve(mainCtx, healthPort)
+		if err != nil {
+			return log.Errorf("Error starting health port, exiting: %v", err)
+		}
+		log.Debugf("Health check listening on port %d", healthPort)
+	}
+
+	// get hostname
+	hostname, err := util.GetHostname()
+	if err != nil {
+		return log.Errorf("Error while getting hostname, exiting: %v", err)
+	}
+	log.Infof("Hostname is: %s", hostname)
+
+	// setup the forwarder
+	keysPerDomain, err := config.GetMultipleEndpoints()
+	if err != nil {
+		log.Error("Misconfiguration of agent endpoints: ", err)
+	}
+	f := forwarder.NewDefaultForwarder(keysPerDomain)
+	f.Start()
+	s := serializer.NewSerializer(f)
+
+	aggregatorInstance := aggregator.InitAggregator(s, nil, hostname, "cluster_agent")
+	aggregatorInstance.AddAgentStartupTelemetry(fmt.Sprintf("%s - Datadog Cluster Agent", version.AgentVersion))
+
+	log.Infof("Datadog Cluster Agent is now running.")
+
+	// Setup a channel to catch OS signals
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+
+	// initialize BBS Cache before starting provider/listener
+	err = initializeBBSCache()
+	if err != nil {
+		return err
+	}
+
+	// create and setup the Autoconfig instance
+	common.SetupAutoConfig(config.Datadog.GetString("confd_path"))
+	// start the autoconfig, this will immediately run any configured check
+	common.StartAutoConfig()
+
+	var clusterCheckHandler *clusterchecks.Handler
+	clusterCheckHandler, err = setupClusterCheck(mainCtx)
+	if err != nil {
+		log.Errorf("Error while setting up cluster check Autodiscovery %v", err)
+	}
+
+	// Start the cmd HTTPS server
+	// We always need to start it, even with nil clusterCheckHandler
+	// as it's also used to perform the agent commands (e.g. agent status)
+	sc := clusteragent.ServerContext{
+		ClusterCheckHandler: clusterCheckHandler,
+	}
+	if err = api.StartServer(sc); err != nil {
+		return log.Errorf("Error while starting agent API, exiting: %v", err)
+	}
+
+	// Block here until we receive the interrupt signal
+	<-signalCh
+
+	// retrieve the agent health before stopping the components
+	// GetStatusNonBlocking has a 100ms timeout to avoid blocking
+	health, err := health.GetStatusNonBlocking()
+	if err != nil {
+		log.Warnf("Cluster Agent health unknown: %s", err)
+	} else if len(health.Unhealthy) > 0 {
+		log.Warnf("Some components were unhealthy: %v", health.Unhealthy)
+	}
+
+	// Cancel the main context to stop components
+	mainCtxCancel()
+
+	log.Info("See ya!")
```

Comment:
Is the familiar wording intended here?

Reply:
Again, this is exactly what DCA for kubernetes has and further refactoring will make it in line with whatever happens with that.